scheduler/scheduler.go

```diff
@@ -210,16 +207,22 @@ func (s *Scheduler) Sync(ctx context.Context, client schema.ClientMeta, tables s
 			panic(fmt.Errorf("unknown scheduler %s", s.strategy.String()))
 		}
 	}()
-	for resource := range resources {
+
+	b := newBatcher(res, 50, 15*time.Second)
+	done := ctx.Done()
+	for {
 		select {
-		case <-ctx.Done():
+		case <-done:
```

Comment:
This optimization is probably unnecessary as `ctx.Done()` will also return the same member every time, and it's hopefully optimized

Reply:
it will return the same value, but the func will call atomic loads, which impacts performance (see cancelCtx implementation, for instance)

Reply:
https://cs.opensource.google/go/go/+/refs/tags/go1.22.2:src/context/context.go;l=438-451